bundles/org.openhab.binding.awattar/src/main/java/org/openhab/binding/awattar/internal/handler/aWATTarBestpriceHandler.java

```diff
@@ -0,0 +1,287 @@
+import org.openhab.binding.awattar.internal.aWATTarPrice;
+import org.openhab.core.i18n.TimeZoneProvider;
+import org.openhab.core.library.types.OnOffType;
+import org.openhab.core.library.types.StringType;
+import org.openhab.core.thing.Bridge;
+import org.openhab.core.thing.Channel;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.binding.BaseThingHandler;
+import org.openhab.core.thing.type.ChannelKind;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.RefreshType;
+import org.openhab.core.types.State;
+import org.openhab.core.types.UnDefType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link aWATTarBestpriceHandler} is responsible for computing the best prices for a given configuration.
+ *
+ * @author Wolfgang Klimt - Initial contribution
+ */
+@NonNullByDefault
+public class aWATTarBestpriceHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(aWATTarBestpriceHandler.class);
+
+    private final int thingRefreshInterval = 60;
+    @Nullable
+    private ScheduledFuture<?> thingRefresher;
+
+    private final TimeZoneProvider timeZoneProvider;
+
+    public aWATTarBestpriceHandler(Thing thing, TimeZoneProvider timeZoneProvider) {
+        super(thing);
+        this.timeZoneProvider = timeZoneProvider;
+    }
+
+    @Override
+    public void initialize() {
+        logger.trace("Initializing aWATTar bestprice handler {}", this);
+        aWATTarBestpriceConfiguration config = getConfigAs(aWATTarBestpriceConfiguration.class);
+        logger.trace("Got Config: {}", config.toString());
+
+        boolean configValid = true;
+
+        if (config.rangeStart < 0 || config.rangeStart > 23) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "@text/error.start.value");
+            configValid = false;
+        }
+        if (config.rangeDuration < 1 || config.rangeDuration > 24) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "@text/error.duration.value");
+            configValid = false;
+        }
+
+        if (config.length < 1 || config.length >= config.rangeDuration) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "@text/error.length.value");
+            configValid = false;
+        }
+
+        if (!configValid) {
+            return;
+        }
+
+        synchronized (this) {
+            ScheduledFuture<?> localRefresher = thingRefresher;
+            if (localRefresher == null || localRefresher.isCancelled()) {
+                logger.trace("Start Thing refresh job at interval {} seconds.", thingRefreshInterval);
+                thingRefresher = scheduler.scheduleAtFixedRate(this::refreshChannels, getMillisToNextMinute(1),
```

Comment:
At this point I need to keep it because the binding should act exactly at minute borders. This is not possible with scheduleWithFixedDelay without requiring further logic to adapt to the "drift" that would come with this approach. At least I was able to change one of the three refreshers I have.